lib/stamp/translator.rb

```diff
@@ -100,7 +99,7 @@ def two_digit_hour_emitter(token)
         Emitters::TwoDigit.new(:hour)
       else
         # 12-hour clock with leading zero
-        Emitters::TwoDigit.new(:hour, &HOUR_TO_12_HOUR)
+        Emitters::TwelveHour.leading_zero
```

Comment:
this is very different from the `TWO_DIGIT_YEAR_EMITTER` approach

Thinking about moving these all up into constants at the top and then memoizing / `leading_zero` from there.

thoughts?